q2_feature_table/_summarize/_visualizer.py

```diff
@@ -155,6 +163,79 @@ def summarize(output_dir: str, table: biom.Table,
         fh.write(');')
 
 
+def _compute_descriptive_stats(lst: list):
+    """Basic descriptive statistics and a (parametric) seven-number summary.
+
+    Calculates descriptive statistics for a list of numerical values, including
+    count, min, max, mean, and a parametric seven-number-summary. This summary
+    includes values for the lower quartile, median, upper quartile, and
+    percentiles 2, 9, 91, and 98. If the data is normally distributed, these
+    seven percentiles will be equally spaced when plotted.
+
+    Parameters
+    ----------
+    lst : list of int or float values
+
+    Returns
+    -------
+    dict
+        a dictionary containing the following descriptive statistics:
+
+        count
+            int: the number of items in `lst`
+        min
+            int or float: the smallest number in `lst`
+        max
+            int or float: the largest number in `lst`
+        mean
+            float: the mean of `lst`
+        seven_num_summ_percentiles
+            list of floats: the parameter percentiles used to calculate this
+            seven-number summary: [2, 9, 25, 50, 75, 91, 98]
+        seven_num_summ_values
+            list of floats: the calculated percentile values of the summary
+
+    """
+    # NOTE: Not built to handle 'lst's containing non-numerical values
+
+    if len(lst) == 0:
+        raise ValueError('No values provided.')
+    count = len(lst)
+    minimum = min(lst)
+    maximum = max(lst)
+    range = maximum-minimum
+    mean = np.mean(lst)
+
+    seven_num_summ_percentiles = [2.0, 9.0, 25.0, 50.0, 75.0, 91.0, 98.0]
+    seven_num_summ_values = np.percentile(
```

Comment:
Check out this function for computing the 7-number summary in pandas:

https://github.com/qiime2/q2-demux/blob/master/q2_demux/_summarize/_visualizer.py#L90-L95.

If you update to use `pandas` below this might make things even easier, since you can stay in pandas for everything.